### PR TITLE
fides - update system.legal_basis_for_transfers and system.legal_basis_for_profiling

### PR DESCRIPTION
**Pull request by adamsachs**
Part of #3974 

‼️ contains migration
‼️ ensure `requirements.txt` is pinned to actual `fideslang` version once released

### Description Of Changes

See issue for context/motivation. 

### Code Changes

* [x] bump `fideslang` dependency to integrate associated updates
* [x] update db models:  update `system.legal_basis_for_transfers` and `system.legal_basis_for_profiling` to be multivalue
* [x] associated migration for above changes: also includes data migration that nests previous single value text field as an array element, and then necessary value updates for `system.legal_basis_for_transfers`

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    # (enum is not enforced/specified on the DB, just API-level)
+
+    for row in existing_values_transfers:
+        existing_value = row[0]
+        updated_value = legal_basis_transfers_upgrades.get(
+            existing_value, existing_value
+        )
+        updated_value = [
+            updated_value
+        ]  # wrap in a list because column is now array type
```
Comment (pattisdr) [inline src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py:73]: Early review comment, apologies if I'm butting in too early - make sure updated_value is not null so you don't get {NULL}

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    )
+
+    # data migration - wrap values in a list
+    for row in existing_values_profiling:
+        existing_value = row[0]
+        updated_value = legal_basis_transfers_upgrades.get(
+            existing_value, existing_value
+        )
+        updated_value = [
+            updated_value
```
Comment (pattisdr) [inline src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py:111]: and here!

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        )
+
+
+def downgrade():
+    """
+    Downgrade from array fields to single-value.
+
+    We do not try to perform a data migration, since we'd have to
+    drop some values arbitrarily. Better to just null the column out
+    to make it clear that manual intervention is needed.
```
Comment (pattisdr) [inline src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py:127]: what do you mean? can we still migrate what we can?

Comment (pattisdr) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py:127]: ah multiple values potentially, right

Comment (pattisdr) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py:127]: maybe a log here

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py (hunk 1 above)
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py:73]: never too early, thank you!!! good catch 🙏

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py (hunk 3 above)
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py]: added in [1857c7b](https://github.com/ethyca/fides/pull/3977/commits/1857c7b398969554f8d0e0192d674813c7ce0614)

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py (hunk 2 above)
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py]: addressed in [1857c7b](https://github.com/ethyca/fides/pull/3977/commits/1857c7b398969554f8d0e0192d674813c7ce0614)

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py (hunk 1 above)
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py]: addressed in [1857c7b](https://github.com/ethyca/fides/pull/3977/commits/1857c7b398969554f8d0e0192d674813c7ce0614) - does that look good to you? seemed to work for me locally on a manual smoke test.

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py (hunk 3 above)
Comment (pattisdr) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py]: thinking more, there's also an argument to be made for keeping the first one? wouldn't most just have one? maybe keep the first and log a list of records where data was lost to reduce downgrade effort. Your call though.

**Review by pattisdr — APPROVED** [review]

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py (hunk 3 above)
Comment (adamsachs) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py]: yeah, that's a fair argument...i had been weighing it a bit.

one other factor that i'm taking into consideration is that this downgrade is relevant for very few use cases - i don't think we've even officially released a version of the software that has these fields but has them as single-valued, i.e. this downgrade logic really applies to intra-release downgrades only. i think that it basically would have no impact outside of our self-hosted internal instances that have gotten incremental upgrades throughout these past couple of weeks, and i think in the (hopefully unlikely) scenario they need to downgrade, we could live with these fields getting wiped?

so ultimately i felt that it wasn't worth figuring out and testing that logic, but maybe i'm just being a bit lazy 😬

Comment (pattisdr) [reply src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py]: no you're right I forgot that it hadn't been released yet, that's fine!

File: src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+            type_=sa.ARRAY(sa.String()),
+            server_default="{}",
+            nullable=True,
+        ),
+    )
+
+    # data migration - wrap values in a list
+    for row in existing_values_profiling:
+        existing_value = row[0]
+        updated_value = legal_basis_transfers_upgrades.get(
```
Comment (pattisdr) [inline src/fides/api/alembic/migrations/versions/3038667ba898_update_system_legal_basis_fields.py:108]: taking another look 👀 

there was this copy paste issue but that shouldn't have had a practical effect.  The profiling value wouldn't have existed in the legal basis for transfers dictionary, so we would have used the original profiling value and added it to the list

**Pull request merged**